cva6/tests/custom/sv32/vm_mstatus_tvm_01.S

```diff
@@ -0,0 +1,176 @@
+#=======================================================================
+#      RW Access of satp and sfence.vma in User and Supervisor mode when 
+#      mstatus.tvm = 1  
+#-----------------------------------------------------------------------
+# Test Description:
+# If mstatus.TVM=1, read and write access to the satp and sfence.vma will 
+# raise illegal instruction exception in S-mode. This test
+# covers the following scenarios in both supervisor and user privilege
+# modes for level1 PTE.
+#                           Level 1
+#   - mstatus.tvm = 1 and test the RW access of satp and sfence in 
+#     in supervisor and user mode
+#=======================================================================
+
+#include "macros.h"
+
+#ifdef smode
+    #define SET_PTE_U 0
+#else
+    #define SET_PTE_U PTE_U
+#endif
+
+#define _MMODE_  "M"
+#define _SUMODE_ "SU"
+
+.text
+.global _start
+
+_start:
+    ALL_MEM_PMP                                                            # PMP permission to all the memory
+    la t1,trap_handler                                                     # loads the address of trap handler 
+    csrw mtvec,t1                                                          # sets the mtvec to trap handler 
+    
+# ----------------LEVEL 1 PTE Setup for load and store test------------
+
+    la a1,vm_en                                                            # loads the address of label vm_en
+    mv a0, a1                                                              # set the VA to PA (identity mapping)
+    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_X | PTE_V )              # sets the permission bits
+    PTE_SETUP_RV32 ( a1, a2, t1, a0, pgtb_l1, LEVEL1 )                     # setup the PTE for level1
+ 
+    la a1,rvtest_data                                                      # loads the address of label rvtest_data
+    mv a0, a1                                                              # set the VA to PA (identity mapping)
+    ori a2, x0, ( PTE_V | PTE_A | SET_PTE_U | PTE_D | PTE_W | PTE_R | PTE_X)# sets the permission bits -- only W permission
+    PTE_SETUP_RV32(a1, a2, t1, a0, pgtb_l1, LEVEL1)                        # setup the PTE for level1   
+
+    la a1,rvtest_check                                                     # loads the address of label rvtest_data
+    mv a0, a1                                                              # set the VA to PA (identity mapping)                                          
+    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_W | PTE_R | PTE_V )      # sets the permission bits
+    PTE_SETUP_RV32(a1, a2, t1, a0, pgtb_l1, LEVEL1)                        # setup the PTE for level1
```

Comment:
Do we need to setup page table for this test?

Reply:
yeah we need this as there will be an exception when on access of satp and sfence